### PR TITLE
Savostin

### PR DESCRIPTION
**Pull request by ESavostin**
Добавил файлы для подстановки в форму отправки CV. В названиях указаны их предназначения для тестов

### CODE REVIEW

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Feature: 1.0.0 buttons lead to correct pages
+Background: 
+	Given "Career Quantori" page is opened
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature:3]: I would recommend to add URL (https://career.quantori.com/) instead of just the name of the website in order to avoid any misunderstanding in the future. :)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 2]
```diff
@@ -0,0 +1,52 @@
+Feature: 1.0.0 buttons lead to correct pages
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature:1]: As I have noticed, you decided to check the main page on Quantori website. I would recommend to specify it in feature name, so that we know what is exactly going on here. :)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+	Examples:
+	|button            |page                                 |
+	|About             |https://career.quantori.com/about    |
+	|Open job positions|https://career.quantori.com/positions|
+	|Contact           |https://career.quantori.com/contact  |
+
+Scenario Outline: 1.0.0_02 All buttons of footer lead to correct pages
+	When user clicks on <button>
+	# VPN must be active
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Why do we need VPN? I can access https://career.quantori.com without VPN

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 1 above)
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: I would recommend to add URL (https://career.quantori.com/) instead of just the name of the website in order to avoid any misunderstanding in the future. :)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 2 above)
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: As I have noticed, you decided to check the main page on Quantori website. I would recommend to specify it in feature name, so that we know what is exactly going on here. :)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 4]
```diff
@@ -0,0 +1,52 @@
+Feature: 1.0.0 buttons lead to correct pages
+Background: 
+	Given "Career Quantori" page is opened
+
+Scenario Outline: 1.0.0_01 All buttons of header lead to correct pages
+	When user clicks on <button>
+	Then <page> is opened
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: After "Then" it is recommended to use "should do/be done" construction, so I'd rephrase it: "Then ... should be opened"

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+	Examples:
+	|button            |page                                 |
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: From my point of view, one more button should be added here: Quantori logo in the left upper corner

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	|About             |https://career.quantori.com/about    |
+	|Open job positions|https://career.quantori.com/positions|
+	|Contact           |https://career.quantori.com/contact  |
+
+Scenario Outline: 1.0.0_02 All buttons of footer lead to correct pages
+	When user clicks on <button>
+	# VPN must be active
+	Then <page> is opened
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Then ... should be opened

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+	Examples:
+	|button         |page                                                                             |
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: I would recommend to add 2 more buttons: Quantori logo in the bottom left corner and "Data Privacy Policy" as these two are also included in footer

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+	|postal envelope|https://mail.google.com/mail/u/0/?fs=1&tf=cm&source=mailto&to=career@quantori.com|
+	|hh             |https://hh.ru/employer/3343960                                                   |
+	|f              |https://www.facebook.com/quantoricom/                                            |
+	|camera         |https://www.instagram.com/quantori_community/                                    |
+	|vk             |https://vk.com/quantori                                                          |
+	|in             |https://www.linkedin.com/company/quantori                                        |
+
+Scenario Outline: 1.0.0_03 All the rest buttons of body lead to correct pages
+	When user clicks on <button>
+	Then <page> is opened
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Then ... should be opened

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+	Examples: 
+	|button                 |page                                 |
+	|More open job positions|https://career.quantori.com/positions|
+	|Learn more             |https://career.quantori.com/about    |
+
+Scenario Outline: 1.0.0_04 Buttons on "Hot open job positions"-table lead to position-pages with the same names
+	When user clicks on <position>
+	Then page with <name> is opened
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Then ... should be opened

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+	|button         |page                                                                             |
+	|planet         |https://www.quantori.com                                                         |
+	|postal envelope|https://mail.google.com/mail/u/0/?fs=1&tf=cm&source=mailto&to=career@quantori.com|
+	|hh             |https://hh.ru/employer/3343960                                                   |
+	|f              |https://www.facebook.com/quantoricom/                                            |
+	|camera         |https://www.instagram.com/quantori_community/                                    |
+	|vk             |https://vk.com/quantori                                                          |
+	|in             |https://www.linkedin.com/company/quantori                                        |
+
+Scenario Outline: 1.0.0_03 All the rest buttons of body lead to correct pages
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: I would recommend to change the name for this scenario, make it more specific. May be, as we only have two buttons, just add the names of these buttons? Additionally, in the next scenario you also check the buttons in the body.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 3 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: We need VPN to open linkedin (the rightmost and lowest button of the page).  This comment about VPN has been updated.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 1 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: You`re right. Done)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 2 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Done!

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 4 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: done)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 5 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Done) You can find it in buttons_of_Career_Quantori_page.feature 1.0.0_01

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 6 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Done

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 7 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Done! You can find it in buttons_of_Career_Quantori_page.feature 1.0.0_02

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 8 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Done!

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 9 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Done)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature (hunk 10 above)
Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons.feature]: Done) I combined the scenarios, so you can see the result in buttons_of_Career_Quantori_page.feature 1.0.0_03

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 11]
```diff
@@ -0,0 +1,179 @@
+Feature: 1.0.0 Sending CV
+Background: 
+	Given "Career Quantori" page is opened
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature]: Same here: I would recommend to add URL (https://career.quantori.com/) instead of just the name of the website in order to avoid any misunderstanding in the future. :)

Comment (ESavostin) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature]: Done)

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+	And only valid CV can be attached to form
+	And all files are in Savostin-folder
+
+Scenario:1.0.0_01 empty form can not be sent
+	When user clicks on "Send" button
+	And all fields are empty
+	Then form is not sent
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:10]: Syntax: 
Then form should not be sent
And following should be displayed ...


General for scenarios 1.0.0_02, 1.0.0_03, 1.0.0_04, 1.0.0_05, 1.0.0_06: 
I would add the names of the fields where the errors appears in a table view: 

| Name field | "invalid name format" |
| Email field | "invalid email format" |

I am not that sure here though. :)

...

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+
+	Examples: 
+	|name            |
+	|�������         |
+	|������� ��������|
+	|Eugene          |
+	|asdadgsgdfm     |
+	|@!#!@#$!@$      |
+
+Scenario Outline:1.0.0_03 form with only valid name on it can not be sent
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:35]: As you have only one example of the name in this scenario outline, I believe it would be better to change it to regular scenario and just add the name somehow so: 

When user clicks on "Send" button
And user types "Eugene Savostin" to the "Name" field

Otherwise, you can add some other examples of the names.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+	When user clicks on "Send" button
+	And all fields are empty except name with <name>
+	Then form should not be sent
+	And following errors should be displayed appropriate fields:
+	|"this field is required"                             |
+	|"attach CV" becomes red                              |
+	|"I agree to the personal data processing" becomes red|
+
+	Example: Eugene Savostin
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:44]: Examples:
|Eugene Savostin|

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+	When user clicks on "Send" button
+	And all fields are empty except CV "valid_file_to_attach.docx" is attached
+	Then form should not be sent
+	And following errors should be displayed appropriate fields:
+	|"this field is required"|
+	|"this field is required"|
+	|"I agree to the personal data processing" becomes red|
+
+Scenario Outline:1.0.0_07 form with all valid data except name on it can not be sent
+	Given field name contains <name>
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:85]: Same here: As you have only one example of the email in this scenario outline, I believe it would be better to change it to regular scenario and just add the name somehow so:

And field email contains " eugene@gmail.com "
...

Otherwise, you can add some other examples of the names and emails.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+	When user clicks on "Send" button
+	Then form should not be sent
+	And following error should be displayed under field "name":
+	|"invalid name format"|
+
+	Examples: 
+	|name  |email           |
+	|Eugene|eugene@gmail.com|
+
+Scenario Outline:1.0.0_08 form with all valid data except email on it can not be sent
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:98]: Same here as in the comment above

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+Scenario:1.0.0_10 form can not take file heavier than 10MB
+	When user chooses file that is heavier than 10MB
+	Then the file should not be taken
+	And the window with following text appears:
+	"""
+	Something went wrong
+	Max file size 10 Mb 
+	"""
+
+Scenario Outline:1.0.0_11 form can not be sent without confirmation of personal data processing
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:130]: Same here about using scenario instead of scenario outline

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 18]
```diff
@@ -0,0 +1,179 @@
...
+	And personal data processing is not confirmed
+	When user clicks on "Send" button
+	Then form should not be sent
+	And "I agree with personal data processing" becomes red
+
+	Examples:
+	|name              |email             |
+	|Alexandr Mikhailov|testmail@gmail.com|
+
+Scenario Outline:1.0.0_12 form with all valid data and confirmation of personal data processing can be sent
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:143]: Same here about using scenario instead of scenario outline

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+	"""
+	invalid email format
+	"""
+	Examples:
+	|email                            |
+	|�����������@mail.ru              |
+	|���������@mail.comcomcomcomcomcom|
+	# Fully Qualified Domain Name forbids more than 15 symbols in domain field of address.
+
+Scenario:1.0.0_14 forms field "name" can contain only 70 characters
```
Comment (IuliiaSat) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:168]: Complicated. :)

May be, it is easier just to add the text how you want it to be added to the "Text" field?

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 20]
```diff
@@ -0,0 +1,179 @@
+Feature: 1.0.0 Sending CV
+Background: 
+	Given https://career.quantori.com page is opened
+	And only valid CV can be attached to form
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:4]: Strings 4-5 excess, because it is not a state of a system.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 21]
```diff
@@ -0,0 +1,179 @@
...
+	And all files are in Savostin-folder
+
+Scenario:1.0.0_01 empty form can not be sent
+	When user clicks on "Send" button
+	And all fields are empty
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:9]: String 9 - is not an action
Valid for all similar mistakes below.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature (hunk 12 above)
Comment (shepelevaolga) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:10]: But why are we using incorrect errors? 
Look to the screenshot and review what happened with a fields without values: 

image

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 22]
```diff
@@ -0,0 +1,179 @@
...
+	Then form is not sent
+	And following are displayed appropriate fields:
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:11]: Not an understandable phrase.
Incorrect messages. 
You should determine fields where you can see errors. 
```
And the following fields should be highlighted with appropriate errors: 
|Field| Error|
|Name| This field is required|
|Last Name| This field is required|
```

This comment is valid for all similar mistakes.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 23]
```diff
@@ -0,0 +1,179 @@
...
+	|"invalid name format"                                |
+	|"invalid email format"                               |
+	|"attach CV" becomes red                              |
+	|"I agree to the personal data processing" becomes red|
+
+Scenario Outline:1.0.0_02 form with only invalid name on it can not be sent
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:17]: Didn't catch it what do you want to check? 
Validation field "Name" ? Or other fields? 
If you would like to check one field and validate it for the incorrect symbols or another one, you should test only one field, other fields should be filled correct.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature (hunk 13 above)
Comment (shepelevaolga) [reply qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:35]: Not necessary to use Examples in this your case. 
Also, you can check validation for all mandatory fields in one case in this way: 

```
Scenario Outline: User can't send a CV without filling one of the mandatory field 
When the user inputs <Name> to the "Name" field
And user inputs <Email> to the "Email" field
And user uploads <Attachment>
And user makes checkbox in <Checkbox State> state
And user clicks on the "Send" button
Then CV should not be sent 
And <Error message> message should be visible 
Examples: 
|Name          | Email               |Checkbox state| Attachment| Error message|
| -                 |test@mail.com| selected| Attatchment1|This field is required|
|Test Name| -                        | selected| Attachment1| This field is required|
```
etc. 

Valid message for all similar mistakes below.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 24]
```diff
@@ -0,0 +1,179 @@
...
+	Then form should not be sent
+	And following error should be displayed under field "email":
+	|"invalid email format"|
+
+	Examples: 
+	|name           |email              |
+	|Marina Sidorova|thisMailIsIncorrect|
+
+Scenario:1.0.0_09 form can not take file with invalid type
+	When user chooses file with invalid type to input
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:113]: All test data should be prepared and used in your test cases.
We can't say - some files should be uploaded. No, only specific files should be uploaded.

This comment is valid for all similar mistakes through whole feature file.

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature [hunk 25]
```diff
@@ -0,0 +1,179 @@
...
+	Then the file should not be taken
+	And the window with following text appears:
+	"""
+	Something went wrong
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/CV_sending.feature:117]: Is only text visible? Or maybe other elements? 
Close or OK buttons?

File: qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons_of_Career_Quantori_page.feature [hunk 26]
```diff
@@ -0,0 +1,48 @@
+Feature: 1.0.0 All buttons of "Career Quantori" page lead to correct pages
```
Comment (shepelevaolga) [inline qa/automation/CareerPortal.Tests.Scenarios/Savostin/buttons_of_Career_Quantori_page.feature:1]: The name of the feature file should be the equal name of the feature inside of file.
Feature: 1.0.1 Header/Footer of Career Portal

Also, a bad practice to try to test everything in one feature file.
The functional of the Career Portal can be separated into several sections : 
- Header 
- Footer
- Filter of positions 
- List of positions
- Learn more section
- Hiring Process
- CV form

And why do you use the same number of features? 
You already used the number 1.0.0

**Review by shepelevaolga — CHANGES_REQUESTED** [review]
Feedback for the Author: 
- Please, read how we should use the "Given" section and Scenarios Outline + Examples
- Use specific test data for your cases (should be prepared) 
- Pay attention to the Titles - should be transparency for all team members and follow numbers/names of Feature files
- Your scenarios should be executed 

Feedback for the Reviewer:
- Be careful about Gherkin Syntax - check more detailed 
- Check expected results in scenarios - all scenarios should be executed.
